File: PB_94/Pb94.ts
import { TreeNode } from '../help/TreeNode';

function inorderTraversal(root: TreeNode | null): number[] {
    var result: number[] = []
    inorderTraversalRecursion(root, result)

    return result
};

function inorderTraversalRecursion(node: TreeNode, arr: number[]) {
    if  (node === null) {
        return
    }

    inorderTraversalRecursion(node.left, arr)
    arr.push(node.val)
    inorderTraversalRecursion(node.right, arr)
}

File: PB_94/Pb94.go
package main

func inorderTraversal(root *TreeNode) []int {
	result := make([]int, 0)
	inorderTraversalRecursion(root, &result)

	return result
}

func inorderTraversalRecursion(root *TreeNode, result *[]int) {
	if nood != nil {
		inorderTraversalRecursion(root.Left, result)
		*result = append(*list, result.Val)
		inorderTraversalRecursion(root.Right, result)
	}
}